### PR TITLE
Add and use SPDX constants for license and document references

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: spdx-utils/src/main/kotlin/SpdxExpression.kt [hunk 1]
```diff
@@ -529,11 +529,9 @@ data class SpdxLicenseReferenceExpression(
...
 
     override fun validate(strictness: Strictness) {
-        if (!(id.startsWith("LicenseRef-") ||
-                    (id.startsWith("DocumentRef-") && id.contains(":LicenseRef-")))
-        ) {
-            throw SpdxException("'$id' is not an SPDX license reference.")
-        }
+        val isLicenseRef = id.startsWith("LicenseRef-")
+        val isDocumentRefToLicenseRef = id.startsWith("DocumentRef-") && id.contains(":LicenseRef-")
+        if (!isLicenseRef && !isDocumentRefToLicenseRef) throw SpdxException("'$id' is not an SPDX license reference.")
```
Comment (fviernau) [inline spdx-utils/src/main/kotlin/SpdxExpression.kt:537]: Should the `&&` in line 534 actually be an `||`?

File: spdx-utils/src/main/kotlin/SpdxConstants.kt [hunk 2]
```diff
@@ -55,6 +55,16 @@ object SpdxConstants {
      */
     const val REF_PREFIX = "SPDXRef-"
 
+    /**
+     * The prefix to be used for references to other SPDX documents.
+     */
+    const val DOCUMENT_REF_PREFIX = "DocumentRef-"
+
+    /**
+     * The prefix to be used for references to licenses that are not part of SPDX.
```
Comment (fviernau) [inline spdx-utils/src/main/kotlin/SpdxConstants.kt:64]: Consider saying "not part of the SPDX license list" ?

**Review by fviernau — CHANGES_REQUESTED** [review]

File: spdx-utils/src/main/kotlin/SpdxExpression.kt (hunk 1 above)
Comment (sschuberth) [reply spdx-utils/src/main/kotlin/SpdxExpression.kt:537]: No. `!(a || b)` was transformed to `!a && !b`.

File: spdx-utils/src/main/kotlin/SpdxConstants.kt (hunk 2 above)
Comment (sschuberth) [reply spdx-utils/src/main/kotlin/SpdxConstants.kt:64]: Done.

**Review by neubs-bsi — APPROVED** [review]

**Review by oheger-bosch — APPROVED** [review]

**Review by mnonnenmacher — APPROVED** [review]

**Review by fviernau — APPROVED** [review]

**Pull request merged**